### PR TITLE
fix for stubs with identical position but different bends.

### PR DESCRIPTION
**Pull request by tschuh**
The current code expects that stubs are identifiable by its position, this is not true in some samples. This PR fixes the wrong assumption.

### CODE REVIEW

**Comment by tschuh** [comment]
@cgsavard and @Chriisbrown you may want test this branch if you can run your samples without issues.

File: L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc
Comment (tomalin) [inline L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc]: Looks fine. Though based on recent reviews, CMS SW group during reviews seems to prefer "else if" in such constructions. e.g. At L132: else if (a.y() != b.y()).

**Comment by tomalin** [comment]
Does this cause crashes when running with Old KF (prompt or displaced hybrid)? If so, we'd need to add it to the central CMSSW PR too. (Though I'd prefer not to, as it will confuse git).

**Comment by tschuh** [comment]
> Does this cause crashes when running with Old KF (prompt or displaced hybrid)? If so, we'd need to add it to the central CMSSW PR too. (Though I'd prefer not to, as it will confuse git).

no

File: L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc
Comment (tschuh) [reply L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc]: better?

Comment (tomalin) [reply L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc]: yes

**Review by tomalin — APPROVED** [review]
I'm happy with this. Ideally @cgsavard or @Chriisbrown should approve it too, before we merge.

**Review by Chriisbrown — APPROVED** [review]
No longer see the error in the samples we were running, happy for this to be merged

**Pull request merged**
